src/p5.dance.js

```diff
@@ -798,5 +807,8 @@ module.exports = class DanceParty {
     if (this.currentFrameEvents.any && this.onHandleEvents) {
       this.onHandleEvents(this.currentFrameEvents);
     }
+
+    /** Expose for testing **/
+    window.__mostRecentDancePartySpriteCount = this.p5_.allSprites.length;
```

Comment:
I'd rather expose `nativeAPI` in Dance.js — it seems like a concern thats tied to our code-dot-org UI tests, not so much this repo.

Reply:
@joshlory 
(1) I don't want to move objects to `window` as it creates object leaks (making it hard to debug)
(2) I want to make it possible to use this info for dance-party unit tests OR for cdo UI tests

Reply:
Re: (1) can we only do so for test environments?

Reply:
maybe? it doesn't appear that any other cdo repo code does this (looking at `window.__TestInterface` as the primary example). We'd need to route environment info in here through a constructor param.

Reply:
Also, I dismissed the `registerValidation()` approach from the cdo side because it would interfere with the real validation code. The dance API only allows one `validationCallback` to be registered at a time. Without changing that, we'd have to then wrap the level's validation code with our own "test hook" that calls the level's validation code, which seems ugly.

Reply:
The DanceParty class itself doesn't need to know if it should be in test mode, I think we should expose the whole `nativeAPI` object like we do for Game Lab.